TriggerRedHatBuild.groovy

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020-present Sonatype, Inc. All rights reserved.
+ * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+@Grab('io.github.http-builder-ng:http-builder-ng-core:1.0.4')
+
+import groovyx.net.http.HttpBuilder
+import groovyx.net.http.HttpException
+
+if (args.size() < 3) {
+  println 'Usage: groovy TriggerRedhatBuild.groovy <version> <projectId> <apiKey>'
+  return
+}
+
+def (version, projectId, apiKey) = args
+
+final nextTag = getNextTag(apiKey, projectId, version)
+println "Deploying as ${nextTag}"
+
+build(apiKey, projectId, nextTag)
+
+final completedBuild = getCompletedBuild(apiKey, projectId, nextTag)
+
+println publish(apiKey, projectId, completedBuild.digest, completedBuild.name)
+
+String getNextTag(String apiKey, String projectId, String version) {
+  final tags = HttpBuilder.configure {
+    request.uri = "https://connect.redhat.com/api/v2/projects/${projectId}/tags"
+    request.headers['Authorization'] = "Bearer ${apiKey}"
+    request.contentType = 'application/json'
+    request.body = [:]
+  }.post().tags*.name.collectMany {
+    it.split(', ').collect()
+  }
+
+  final currentIndex = tags.findAll {
+    it.startsWith(version)
+  }.collect {
+    it.replaceAll(/${version}-(\d+)-?.*/, '$1') as Integer
+  }.sort().reverse()[0]
+
+  final nextIndex =((currentIndex ?: 0) as Integer) + 1
+
+  return "${version}-${nextIndex}"
+}
+
+Map build(String apiKey, String projectId, String nextTag) {
+  return HttpBuilder.configure {
+    request.uri = "https://connect.redhat.com/api/v2/projects/${projectId}/build"
+    request.headers['Authorization'] = "Bearer ${apiKey}"
+    request.contentType = 'application/json'
+    request.body = [tag: nextTag]
+  }.post()
+}
+
+Map getCompletedBuild(String apiKey, String projectId, String nextTag) {
+  while (true) {
+    println 'Waiting for build to finish.'
+    sleep 60000
+
+    final newTags = HttpBuilder.configure {
```

Comment:
looks like a repeated method we could extract?

Reply:
👍 this simple script is turning into a little architectural marvel. that'll be a nice little closure that can be passed into the methods that use it.

Reply:
this is a little more OO now, with the methods being called normally without any function pointers or that jazz.